### PR TITLE
Warning pop-up window has been created

### PR DESCRIPTION
**Pull request by luchward**
**Previous result:**
Warning message does not appear when user does not save changes and goes to another page, [bug#1320](https://github.com/ita-social-projects/GreenCity/issues/1320)
The system doesn't display notification when a user wants to delete its comment/reply, [bug#1631](https://github.com/ita-social-projects/GreenCity/issues/1631)

**Actual result:**

1. Previously existing pop-up component has been removed as it does not meet SOLID principles.
2. Shared warning-pop-up component has been created; it has common interface and it is used in delete-comment, reply-comment, create-news, edit-profile components.
3. Pending Changes Guard has been created which requires confirmation to leave a page.
4. Parent form-base component has been created, that implements CanDeactivate method to prevent a user leaves the page without saving data; create-news and edit-profile components inherit from form-base component.

98447493-dd00e380-212d-11eb-8868-222af2ae9529

### CODE REVIEW

File: src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.scss [hunk 1]
```diff
@@ -64,6 +68,11 @@ $main-font: 'Open Sans', sans-serif;
   text-align: center;
 }
 
+.edit-news:hover {
+  border-color: #056b33;
```
Comment (skrebetsnataliia) [inline src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.scss:72]: I'm not sure but, maybe better to use variable

Comment (luchward) [reply src/app/component/eco-news/components/eco-news-detail/eco-news-detail.component.scss:72]: It should be done for the whole file, but it is not the aim of this PR.

**Review by skrebetsnataliia — APPROVED** [review]

File: src/app/component/comments/components/delete-comment/delete-comment.component.ts [hunk 2]
```diff
@@ -9,16 +11,35 @@ import { CommentsDTO } from '../../models/comments-model';
...
+
+    const subscribeDialog = dialogRef.afterClosed().subscribe(confirm => {
+      if (confirm) {
+        this.commentsService.deleteComments(this.element.id).subscribe(response => {
+          if (response.status === 200) {
+            this.elementsList.emit();
+          }
+        });
+      }
+      subscribeDialog.unsubscribe();
```
Comment (JimHolman1199) [inline src/app/component/comments/components/delete-comment/delete-comment.component.ts:42]: if i understand this correctly, is subscribeDialog designed to initialize a component? 
you subscribe, get one value and then you immediately unsubscribe ? 
if yes then consider using : take(1) or takeUntil() operators

File: src/app/component/eco-news/components/create-edit-news/create-edit-news.component.ts [hunk 3]
```diff
@@ -37,27 +37,67 @@ export class CreateEditNewsComponent implements OnInit, OnDestroy {
...
     this.onSourceChange();
-    this.isFormInvalid = !this.form.valid ||
-                          this.isArrayEmpty ||
-                          !this.isLinkOrEmpty ||
-                          this.isImageValid();
+  }
+
+  public setInitialValues(): void {
+    if (!this.createEcoNewsService.isBackToEditing) {
+      this.initialValues = this.form.value;
```
Comment (JimHolman1199) [inline src/app/component/eco-news/components/create-edit-news/create-edit-news.component.ts:80]: use getFormValues() insted of this.form.value
or ?

File: src/app/component/eco-news/components/create-edit-news/create-edit-news.component.ts [hunk 4]
```diff
@@ -75,13 +115,15 @@ export class CreateEditNewsComponent implements OnInit, OnDestroy {
...
       }
+      this.setInitialValues();
     } else {
       if (this.newsId) {
         this.fetchNewsItemToEdit();
         this.setDataForEdit();
       } else {
         this.form = this.createEditNewsFormBuilder.getSetupForm();
         this.setDataForCreate();
+        this.setInitialValues();
```
Comment (JimHolman1199) [inline src/app/component/eco-news/components/create-edit-news/create-edit-news.component.ts:126]: not a problem at all, but what about dividing this logic into different methods? or come up with something better than if -else 
https://dev.to/damxipo/avoid-use-if-on-our-js-scripts-1b95

**Review by JimHolman1199 — APPROVED** [review]

**Review by JimHolman1199 — APPROVED** [review]

**Review by vitaliipolitylo — APPROVED** [review]

**Review by Benerous — APPROVED** [review]

**Review by romanovskyroma — APPROVED** [review]

**Pull request merged**